locator/abstract_replication_strategy.cc

```diff
@@ -136,7 +147,7 @@ insert_token_range_to_sorted_container_while_unwrapping(
 }
 
 dht::token_range_vector
-effective_replication_map::do_get_ranges(noncopyable_function<bool(inet_address_vector_replica_set)> should_add_range) const {
+effective_replication_map::do_get_ranges(noncopyable_function<stop_iteration(bool&, const inet_address&)> should_add_range) const {
```

Comment:
I think your rewrite of get_ranges() and do_get_ranges() using the new for_each_natural_enpoint_until made the implementation very hard to understand - e.g., you left the functor name "should_add_range" but it's no longer a bool ("should").

Reply:
I'll try to come up with a better name for the functor parameter.
How about `should_add_range_to_endpoint`?

Reply:
No, my problem is with the word "should_add". It suggests the functor should return a boolean of whether to add something - not a boolean of whether to stop the iteration.

Reply:
Then `consider_add_range_to_endpoint` maybe?